### PR TITLE
Fix the the doxygen text for cudf::concatenate and other places

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Fixes doxygen for `cudf::concatenate`. Also fixes a surprising number of  occurrences of `the the` in comments in cudf source files. And also adds a missing group to the `doxygen_groups.h` header file.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by galipremsagar — APPROVED** [review]

File: cpp/include/cudf/concatenate.hpp [hunk 1]
```diff
@@ -34,25 +34,25 @@ namespace cudf {
...
+ * `views` into an `rmm::device_buffer`
  *
- * Returns empty `device_buffer` if the column is not nullable
+ * Returns an empty buffer if the column is not nullable.
  *
- * @param views host_span of column views whose bitmasks will be concatenated
- * @param mr Device memory resource used for allocating the new device_buffer
- * @return A `device_buffer` containing the bitmasks of all the column views in the views vector
+ * @param views Column views whose bitmasks will be concatenated
+ * @param mr Device memory resource used for allocating return memory
```
Comment (bdice) [inline cpp/include/cudf/concatenate.hpp:42]:
```suggestion
 * @param mr Device memory resource used for allocating the returned memory
```

**Review by bdice — APPROVED** [review]
Looks good to me. Thanks for the the fix!

**Review by harrism — APPROVED** [review]

**Comment by karthikeyann** [comment]
is it possible to add "the the" to codespell as a special rule?

**Review by karthikeyann — APPROVED** [review]
Nice find!

**Comment by harrism** [comment]
> I would not want to start such a trend myself.

You wouldn't be starting it. Most spell checkers I have used flag repeated articles. 
image

**Review by ttnghia — APPROVED** [review]

**Review by gerashegalov — APPROVED** [review]

**Pull request merged**